client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/utils/style-variations.ts

```diff
@@ -14,3 +14,20 @@ export function removeLegacyDesignVariations( designs: Design[] ) {
 		);
 	return designs.filter( ( design ) => ! isVariationOfAnotherDesign( design ) );
 }
+
+/**
+ * Promote designs with style variations by having them at the beginning of the list.
+ */
+export function promoteDesignVariations( designs: Design[] ) {
+	return designs.sort( ( a, b ) => {
+		if ( ! a.style_variations || a.style_variations.length === 0 ) {
```

Comment:
This comparator does not satisfy the requirements for `sort()` as the "Reflexive" property is broken: if a and b both have variations, or both do not have variations, the function should return 0. (ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)

> Reflexive: compareFn(a, a) === 0.

However, in this implementation, it returns either 1 or -1. It might cause the designs to be re-shuffled from their original shuffled order from wpcom.

(BTW, I personally liked the previous implementation though, where we just concatenating filtered arrays. It is much easier to understand)

Reply:
+1, I always have to stop and think when I encounter this type of sorting 😅

Reply:
😂